### PR TITLE
Add update-tags release process

### PR DESCRIPTION
**Pull request by paulb777**
Port the update tag process from update-versions.sh

Add a verification step to insure that published Firebase tags do not get deleted.

### CODE REVIEW

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift [hunk 1]
```diff
@@ -20,10 +20,10 @@ import FirebaseManifest
 import Utils
 
 enum Tags {
-  static func create(gitRoot: URL) {
+  static func create(gitRoot: URL, afterDelete: Bool = false) {
```
Comment (maksymmalyhin) [inline ZipBuilder/Sources/FirebaseReleaser/Tags.swift:23]: optional: I would prefer more verbose name,  something like `createTags(gitRoot:deleteExistingTags:)`.

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift [hunk 2]
```diff
@@ -36,12 +36,43 @@ enum Tags {
...
+    if afterDelete {
+      verifyDeleteIsOK()
+      Shell.executeCommand("git tag --delete \(tag)", workingDir: gitRoot)
+      Shell.executeCommand("git push --delete origin \(tag)", workingDir: gitRoot)
+    }
     Shell.executeCommand("git tag \(tag)", workingDir: gitRoot)
     Shell.executeCommand("git push origin \(tag)", workingDir: gitRoot)
   }
+
+  private static func verifyDeleteIsOK() {
```
Comment (maksymmalyhin) [inline ZipBuilder/Sources/FirebaseReleaser/Tags.swift:57]: optional: I could not realize what the method does until I read the implementation and usage. A name like `verifyTagsAreSafeToDelete` looks a bit more informative to me.

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift [hunk 3]
```diff
@@ -36,12 +36,43 @@ enum Tags {
...
+    var homeDirURL: URL
+    if #available(OSX 10.12, *) {
+      homeDirURL = FileManager.default.homeDirectoryForCurrentUser
+    } else {
+      fatalError("Run on at least macOS 10.12")
+    }
+    let manifest = FirebaseManifest.shared
+    let firebasePublicURL = homeDirURL.appendingPathComponents(
+      [".cocoapods", "repos", "cocoapods", "Specs", "0", "3", "5", "Firebase"]
```
Comment (maksymmalyhin) [inline ZipBuilder/Sources/FirebaseReleaser/Tags.swift:66]: I curious where the number components come from?

File: ZipBuilder/Sources/FirebaseReleaser/main.swift [hunk 4]
```diff
@@ -41,6 +41,11 @@ struct FirebaseReleaser: ParsableCommand {
           help: "Initialize the release branch")
   var pushOnly: Bool
 
+  /// Set this option to update tags only.
+  @Option(default: false,
+          help: "Update the tags")
+  var updateTags: Bool
```
Comment (maksymmalyhin) [inline ZipBuilder/Sources/FirebaseReleaser/main.swift:47]: Maybe `updateTagsOnly` to be consistent with `pushOnly`  and corresponding help string?

**Review by maksymmalyhin — APPROVED** [review]
👍

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift (hunk 1 above)
Comment (paulb777) [reply ZipBuilder/Sources/FirebaseReleaser/Tags.swift]: Done

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift (hunk 2 above)
Comment (paulb777) [reply ZipBuilder/Sources/FirebaseReleaser/Tags.swift]: Done

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift (hunk 3 above)
Comment (paulb777) [reply ZipBuilder/Sources/FirebaseReleaser/Tags.swift:68]: It's the CocoaPods hash of the word Firebase. It's how CocoaPods does sharding.

File: ZipBuilder/Sources/FirebaseReleaser/main.swift (hunk 4 above)
Comment (paulb777) [reply ZipBuilder/Sources/FirebaseReleaser/main.swift]: Done.

**Review by paulb777 — COMMENTED** [review]
Thanks!

File: ZipBuilder/Sources/FirebaseReleaser/Tags.swift [hunk 5]
```diff
@@ -36,12 +38,43 @@ enum Tags {
...
+      verifyTagsAreSafeToDelete()
+      Shell.executeCommand("git tag --delete \(tag)", workingDir: gitRoot)
+      Shell.executeCommand("git push --delete origin \(tag)", workingDir: gitRoot)
+    }
     Shell.executeCommand("git tag \(tag)", workingDir: gitRoot)
     Shell.executeCommand("git push origin \(tag)", workingDir: gitRoot)
   }
+
+  private static func verifyTagsAreSafeToDelete() {
+    var homeDirURL: URL
```
Comment (ryanwilson) [inline ZipBuilder/Sources/FirebaseReleaser/Tags.swift:60]: Should we be extra safe and run a `pod update` here first to be sure that the local repo is up to date? I imagine it's a very very edge case but maybe something worth doing.

Comment (paulb777) [reply ZipBuilder/Sources/FirebaseReleaser/Tags.swift:60]: Good catch! See #6774

**Pull request merged**